lib/versioning/semver-cargo/index.js

```diff
@@ -0,0 +1,39 @@
+const semver = require('../semver');
+
+function convertToCaret(item) {
+  if (semver.isVersion(item)) {
+    return '^' + item.trim();
+  }
+  return item.trim();
+}
+
+// FIXME: Partial versions like '1.2' don't get converted to '^1.2'
+// NOTE: This might be correct behaviour.
+function cargo2npm(input) {
+  let versions = input.split(',');
+  versions = versions.map(convertToCaret);
+  return versions.join('||');
+}
+
+const isLessThanRange = (version, range) =>
+  semver.isLessThanRange(version, cargo2npm(range));
+
+const isValid = input => semver.isValid(cargo2npm(input));
+
+const matches = (version, range) => semver.matches(version, cargo2npm(range));
+
+const maxSatisfyingVersion = (versions, range) =>
+  semver.maxSatisfyingVersion(versions, cargo2npm(range));
+
+const minSatisfyingVersion = (versions, range) =>
+  semver.minSatisfyingVersion(versions, cargo2npm(range));
+
+module.exports = {
+  ...semver,
+  isLessThanRange,
+  isValid,
+  matches,
+  maxSatisfyingVersion,
+  minSatisfyingVersion,
+  cargo2npm,
+};
```

Comment:
The "missing" functions from semver's interface are:

```
  equals,
  getMajor,
  getMinor,
  getPatch,
  isCompatible,
  isGreaterThan,
  isSingleVersion,
  isStable,
  isVersion,
  getNewValue,
  sortVersions,
```

I think you need to rewrite `isSingleVersion` because in npm "1.2.3" is a single version but in cargo it is not (it is same as "^1.2.3").

And the very important missing one is `getNewValue`. It can be modelled on how semver-composer's works: https://github.com/renovatebot/renovate/blob/4a19d57c6bb66139fb7f678366fe031b36d3a16b/lib/versioning/semver-composer/index.js#L57-L120